programs/server/config.xml

```diff
@@ -371,16 +370,40 @@
                 server - one of LDAP server names defined in 'ldap_servers' config section above.
                         This parameter is mandatory and cannot be empty.
                 roles - section with a list of locally defined roles that will be assigned to each user retrieved from the LDAP server.
-                        If no roles are specified, user will not be able to perform any actions after authentication.
-                        If any of the listed roles is not defined locally at the time of authentication, the authenthication attempt
-                         will fail as if the provided password was incorrect.
+                        If no roles are specified here or assigned during role mapping (below), user will not be able to perform any
+                         actions after authentication.
+                role_mapping - section with LDAP search parameters and mapping rules.
+                        The list of strings (values of attributes) returned by the search will be transformed and the resulting strings
+                         will be treated as local role names and assigned to the user.
+                        There can be multiple 'role_mapping' sections defined inside the same 'ldap' section. All of them will be
+                         applied.
+                    base_dn - template used to construct the base DN for the LDAP search.
+                            The resulting DN will be constructed by replacing all '{username}' and '{bind_dn}' substrings
+                             of the template with the actual user name and bind DN during each LDAP search.
+                    attribute - attribute name whose values will be returned by the LDAP search.
+                    scope - scope of the LDAP search.
+                            Accepted values are: 'base', 'one_level', 'children', 'subtree' (the default).
+                    search_filter - template used to construct the search filter for the LDAP search.
+                            The resulting filter will be constructed by replacing all '{username}', '{bind_dn}', and '{base_dn}'
+                             substrings of the template with the actual user name, bind DN, and base DN during each LDAP search.
+                            Note, that the special characters must be escaped properly in XML.
+                    prefix - prefix, that will be expected to be in front of each string in the original list of strings returned by
```

Comment:
What if the attribute doesn't have a specified prefix in its value? It won't be used as a role name then? I'm not sure but maybe it's better to rename `prefix` into `remove_prefix` to emphasize the fact we remove the prefix (and we don't insert the prefix).

Reply:
As we learned from the feedback from people who routinely configure such things for enterprise, it's a common (and effectively the only) way of mapping the role names, so I think it's better to keep it `prefix` since the underlying functionality is quite familiar to everyone.